pkg/origin/response.go

```diff
@@ -0,0 +1,61 @@
+	Status416           int `json:"status_416"`               // Number of responses received with status code 416 (Range Not Satisfiable) from origin.
+	Status4xx           int `json:"status_4xx"`               // Number of 4xx "Client Error" codes delivered from origin.
+	Status500           int `json:"status_500"`               // Number of responses received with status code 500 (Internal Server Error) from origin.
+	Status501           int `json:"status_501"`               // Number of responses received with status code 501 (Not Implemented) from origin.
+	Status502           int `json:"status_502"`               // Number of responses received with status code 502 (Bad Gateway) from origin.
+	Status503           int `json:"status_503"`               // Number of responses received with status code 503 (Service Unavailable) from origin.
+	Status504           int `json:"status_504"`               // Number of responses received with status code 504 (Gateway Timeout) from origin.
+	Status505           int `json:"status_505"`               // Number of responses received with status code 505 (HTTP Version Not Supported) from origin.
+	Status5xx           int `json:"status_5xx"`               // Number of 5xx "Server Error" codes delivered from origin.
+	Latency0to1         int `json:"latency_0_to_1ms"`         // Number of responses from origin with latency between 0 and 1 millisecond.
+	Latency1to5         int `json:"latency_1_to_5ms"`         // Number of responses from origin with latency between 1 and 5 milliseconds.
+	Latency5to10        int `json:"latency_5_to_10ms"`        // Number of responses from origin with latency between 5 and 10 milliseconds.
+	Latency10to50       int `json:"latency_10_to_50ms"`       // Number of responses from origin with latency between 10 and 50 milliseconds.
+	Latency50to100      int `json:"latency_50_to_100ms"`      // Number of responses from origin with latency between 50 and 100 milliseconds.
+	Latency100to250     int `json:"latency_100_to_250ms"`     // Number of responses from origin with latency between 100 and 250 milliseconds.
+	Latency250to500     int `json:"latency_250_to_500ms"`     // Number of responses from origin with latency between 250 and 500 milliseconds.
+	Latency500to1000    int `json:"latency_500_to_1000ms"`    // Number of responses from origin with latency between 500 and 1,000 milliseconds.
+	Latency1000to5000   int `json:"latency_1000_to_5000ms"`   // Number of responses from origin with latency between 1,000 and 5,000 milliseconds.
+	Latency5000to10000  int `json:"latency_5000_to_10000ms"`  // Number of responses from origin with latency between 5,000 and 10,000 milliseconds.
+	Latency10000to60000 int `json:"latency_10000_to_60000ms"` // Number of responses from origin with latency between 10,000 and 60,000 milliseconds.
+	Latency60000plus    int `json:"latency_60000ms"`          // Number of responses from origin with latency of 60,000 milliseconds and above.
```

Comment:
Aside: I hope that in the future we will have an endpoint to list all available metrics, which we would then be able to use here instead of hard-coding structs.

Reply:
I wish it were possible! Alas, the mapping of the Origin Inspector metrics to viable Prometheus metrics isn't well-defined. In fact the _raison d'être_ of the fastly-exporter is to do exactly the sort of rote translation on display here, which (in Go) more or less demands concrete types like these.